src/Engine/RulesState.php

```diff
@@ -163,6 +217,7 @@ public function autoSave() {
         $typed_data->getRoot()->getValue()->save();
       }
     }
+    return $this;
   }
 
```

Comment:
The method setTypedDataManager() is missing to complete the setter injection pattern here when somebody wants to use/test this with a different typed data manager.